src/states/node.rs

```diff
@@ -2458,3 +2476,336 @@ fn create_first_section_info(public_id: PublicId) -> Result<SectionInfo, Routing
         err
     })
 }
+
+#[cfg(all(test, feature = "mock_parsec"))]
+mod tests {
```

Comment:
I would move all these tests to a separate file (say `src/states/node_tests.rs` or `src/states/node/tests.ts`). This one is already too long.

Reply:
1- Are you ok for it to be a follow up PR?
2- Which version exactly do we want? I would want it to have the same level of access as the current test have.

Reply:
Hm good point about the access. I think `tests` needs to remain a submodule of `node`, so that means `src/states/node/tests.rs`. Don't mind if it's done in a different PR.